### PR TITLE
Save an openvino config summarizing all information related to quantization when saving model

### PR DESCRIPTION
**Pull request by echarlaix**
When quantization is applied with `load_in_8bit=True` or by  specifying a `quantization_argument`, an `openvino_config.json` file will be saved when the model  / model's config is saved through the `save_pretrained` method

### CODE REVIEW

File: docs/source/optimization_ov.mdx
Comment (helena-intel) [inline docs/source/optimization_ov.mdx:1]: Do I understand correctly that with these changes you need to add a quantization_config to .from_pretrained when loading a model, but that doesn't actually do anything until you call quantizer.quantize? 
What is the reason for that change? If the motivation is to save the config, it seems we could just update that after quantizing the model?

File: setup.py [hunk 1]
```diff
@@ -39,18 +39,11 @@
...
-        "openvino>=2023.3",
-        "onnx",
-        "onnxruntime",
-    ],
+    "neural-compressor": ["neural-compressor>=2.2.0", "onnx", "onnxruntime<1.15.0"],
+    "openvino": ["openvino>=2023.3", "onnx", "onnxruntime"],
     "openvino-tokenizers": ["openvino-tokenizers[transformers]"],
-    "nncf": ["nncf>=2.8.1"],
+    # "nncf": ["nncf>=2.8.1"],
+    "nncf": ["nncf @ git+https://github.com/openvinotoolkit/nncf.git"],
```
Comment (echarlaix) [inline setup.py:46]: (will be removed before merging)

File: docs/source/optimization_ov.mdx
Comment (echarlaix) [reply docs/source/optimization_ov.mdx:1]: There is no modification of the quantization process in this PR (you can still apply quantization by providing the `quantization_config` or `load_in_8bit` arguments when creating an  instance of `OVModel`), the main addition is that we save the configuration containing all the information related to quantization when saving the model with `model.save_pretrained(output_dir)`. I also removed the default compression value that was default when creating an `OVConfig` instance as it's not used (we should start moving all quantization parameters in `quantization_config` in the future).

File: docs/source/optimization_ov.mdx [hunk 2]
```diff
@@ -38,8 +38,6 @@ save_dir = "ptq_model"
 def preprocess_function(examples, tokenizer):
     return tokenizer(examples["sentence"], padding="max_length", max_length=128, truncation=True)
 
-# Load the default quantization configuration detailing the quantization we wish to apply
-quantization_config = OVConfig()
```
Comment (echarlaix) [inline docs/source/optimization_ov.mdx:42]: removed as currently unused 

https://github.com/huggingface/optimum-intel/blob/v1.15.2/optimum/intel/openvino/quantization.py#L268

**Comment by AlexKoff88** [comment]
@nikita-savelyevv, @l-bat as they plan to introduce relevant changes as well.

File: optimum/intel/openvino/modeling_base.py [hunk 3]
```diff
@@ -203,12 +217,28 @@ def _from_pretrained(
...
+        return cls(
+            model,
+            config=config,
+            model_save_dir=model_cache_path.parent,
+            quantization_config=quantization_config,
+            **kwargs,
+        )
+
+    @staticmethod
+    def _prepare_quantization_config(
```
Comment (AlexKoff88) [inline optimum/intel/openvino/modeling_base.py:232]: it s probably worth calling it as `_prepare_weight_quantization_config`

**Review by AlexKoff88 — APPROVED** [review]

File: optimum/intel/openvino/modeling_base.py (hunk 3 above)
Comment (echarlaix) [reply optimum/intel/openvino/modeling_base.py]: updated !

**Pull request merged**